### PR TITLE
fix(agent): don't instrument Monolog's $context

### PR DESCRIPTION
**Pull request by misiekn**
This PR disables `$context` argument of `Monolog\Logger::addRecord` instrumentation. 
Context is no longer appended to the message in the log record. The code instrumenting
context can be re-enabled at compile time based on the  `HAVE_CONTEXT_IN_MESSAGE` 
macro value:
* 0 - `$context` argument will not be instrumented: its existence and value are ignored 
* 1 - the message of the log record forwarded by the agent will include the value of `$context`, which is appended to the value of `$message`.

### CODE REVIEW

File: agent/lib_monolog.c [hunk 1]
```diff
@@ -295,17 +310,23 @@ static char* nr_monolog_get_context(const size_t argc,
  */
 static char* nr_monolog_build_message(const size_t argc,
                                       NR_EXECUTE_PROTO TSRMLS_DC) {
+#if !HAVE_CONTEXT_IN_MESSAGE
+  /* Make the compiler happy - argc is not used when $context is ignored */
+  (void)argc;
+#endif
```
Comment (ZNeumann) [inline agent/lib_monolog.c:316]:
```suggestion
static char* nr_monolog_build_message(const size_t argc NRUNUSED, /* unused when $context is ignored */
                                      NR_EXECUTE_PROTO TSRMLS_DC) {
```

Comment (ZNeumann) [reply agent/lib_monolog.c:316]: I've never used NRUNUSED, but I think this should work?

Comment (misiekn) [reply agent/lib_monolog.c:316]: I was thinking about `NRUNUSED` attribute in argument list option but IMHO 
```
#if !HAVE_CONTEXT_IN_MESSAGE
  /* Make the compiler happy - argc is not used when $context is ignored */
  (void)argc;
#endif
```
is more explicit than having `NRUNUSED` attribute on an argument that can be used.

Comment (ZNeumann) [reply agent/lib_monolog.c:316]: I'm okay with either. NRUNUSED is cleaner but the preprocessor is more explicit.

**Review by zsistla — APPROVED** [review]

**Pull request merged**